internal/config/config.go

```diff
@@ -248,47 +267,71 @@ func normalizeFunc(f *flag.FlagSet, name string) flag.NormalizedName {
 	return flag.NormalizedName(name)
 }
 
-func getLog() *Log {
+func resolveLog() *Log {
 	return &Log{
 		Level: viperInstance.GetString(LogLevelKey),
 		Path:  viperInstance.GetString(LogPathKey),
 	}
 }
 
-func getDataPlaneConfig() *DataPlaneConfig {
+func resolveDataPlaneConfig() *DataPlaneConfig {
 	return &DataPlaneConfig{
 		Nginx: &NginxDataPlaneConfig{
-			ReloadMonitoringPeriod: viperInstance.GetDuration(DataPlaneConfigNginxReloadMonitoringPeriodKey),
-			TreatWarningsAsError:   viperInstance.GetBool(DataPlaneConfigNginxTreatWarningsAsErrorsKey),
+			ReloadMonitoringPeriod: viperInstance.GetDuration(NginxReloadMonitoringPeriodKey),
+			TreatWarningsAsError:   viperInstance.GetBool(NginxTreatWarningsAsErrorsKey),
 		},
 	}
 }
 
-func getClient() *Client {
+func resolveClient() *Client {
 	return &Client{
 		Timeout:             viperInstance.GetDuration(ClientTimeoutKey),
 		Time:                viperInstance.GetDuration(ClientTimeKey),
 		PermitWithoutStream: viperInstance.GetBool(ClientPermitWithoutStreamKey),
 	}
 }
 
-func getConfigDir() string {
-	return viperInstance.GetString(ConfigDirectoriesKey)
-}
+func resolveCollector(allowedDirs []string) (*Collector, error) {
+	// We do not want to return a sentinel error because we are joining all returned errors
+	// from config resolution and returning them without pattern matching.
+	// nolint: nilnil
+	if !viperInstance.IsSet(CollectorRootKey) {
+		return nil, nil
+	}
 
-func getMetrics() *Metrics {
-	if !viperInstance.IsSet(MetricsRootKey) {
-		return nil
+	var (
+		err         error
+		exporters   []Exporter
+		receivers   []Receiver
+		healthCheck ServerConfig
+	)
+
+	err = errors.Join(
+		err,
+		resolveMapStructure(CollectorExportersKey, &exporters),
+		resolveMapStructure(CollectorReceiversKey, &receivers),
+		resolveMapStructure(CollectorHealthzEndpointKey, &healthCheck),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal collector config: %w", err)
+	}
+
+	col := &Collector{
+		ConfigPath:      viperInstance.GetString(CollectorConfigPathKey),
+		Exporters:       exporters,
+		Receivers:       receivers,
+		HealthzEndpoint: &healthCheck,
```

Comment:
I'd prefer if we had a list of processors here... But I don't mind leaving this as a top level item if we always have it. I do mind the naming :) Perhaps it should be called Health?

Reply:
Health works, it's a quick rename

Reply:
Renamed to `Health`  and added Processor support (only "batch" currently supported).